openedx/adg/lms/applications/tests/test_admin/test_user_application_adg_admin.py

```diff
@@ -235,26 +246,56 @@ def test_changeform_view_post_with_status(
     }
     UserApplicationADGAdmin.changeform_view(user_application_adg_admin_instance, request, application_id)
 
-    mock_save_application_review_info.assert_called_once_with(user_application, request, NOTE)
+    mock_save_application_status.assert_called_once_with(user_application, request.POST.get('status'))
+    mock_save_admin_notes.assert_called_once_with(user_application, request.user, NOTE)
+    mock_save_message_for_applicant.assert_called_once_with(user_application, request.user, TEST_MESSAGE_FOR_APPLICANT)
     mock_send_application_status_update_email.assert_called_once_with(user_application, TEST_MESSAGE_FOR_APPLICANT)
     mock_changeform_view.assert_called_once_with(request, application_id, extra_context=None)
 
 
 @pytest.mark.django_db
-def test_save_application_review_info(request, user_application):
+def test_save_application_status(user_application):
     """
-    Test that application review information is successfully saved.
+    Test that application status is successfully saved.
     """
-    request.user = UserFactory()
-    request.POST = {'status': UserApplication.ACCEPTED}
-
-    UserApplicationADGAdmin._save_application_review_info('self', user_application, request, NOTE)
+    UserApplicationADGAdmin._save_application_status('self', user_application, UserApplication.ACCEPTED)
 
     updated_application = UserApplication.objects.get(id=user_application.id)
 
     assert updated_application.status == UserApplication.ACCEPTED
-    assert updated_application.internal_admin_note == NOTE
-    assert updated_application.reviewed_by == request.user
+
+
+@pytest.mark.django_db
+def test_save_admin_notes(user_application):
+    """
+    Test that the admin note against an application is successfully saved.
+    """
+    user = UserFactory()
+    UserApplicationADGAdmin._save_admin_notes('self', user_application, user, NOTE)
+
+    admin_note = AdminNote.objects.get(user_application=user_application, saved_by=user, note=NOTE)
+
+    assert admin_note
```

Comment:
I think we should update it like: `assert AdminNote.objects.filter(user_application=user_application, saved_by=user, note=NOTE).exists()`.
Else if some day the `_save_admin_note` gets updated then this test will raise error instead of failing.

Reply:
I tried it with get method as well and the test failed.